include/odp/api/spec/packet.h

```diff
@@ -2140,6 +2140,50 @@ uint32_t odp_packet_payload_offset(odp_packet_t pkt);
  */
 int odp_packet_payload_offset_set(odp_packet_t pkt, uint32_t offset);
 
+/**
+ * Enable or disable Tx packet drop based on age
+ *
+ * Enable or disable Tx packet drop based on packet age. When enabled, packet will be dropped
+ * if it is in Tx pktout queue for longer than per-packet timeout.
+ *
+ * @param pkt     Packet handle
+ * @param enable  True indicates aging to be enabled on the packet.
+ */
+void odp_packet_aging_drop_set(odp_packet_t pkt, odp_bool_t enable);
+
+/**
+ * Set Tx packet aging timeout
+ *
+ * Set a packet aging timeout in usec which limits the age of packet in pktout queue or
+ * traffic scheduler/shapers. Packet exceeding its timeout will be dropped when packet aging
+ * drop is enabled.
+ *
+ * @param pkt     Packet handle
+ * @param tmo_us  Timeout to age in usec
+ *
+ * @see odp_pktio_capability_t::max_tx_aging_tmo_us
+ */
+void odp_packet_aging_tmo_set(odp_packet_t pkt, unsigned int tmo_us);
```

Comment:
Unit of the timeout should be nanoseconds as that is the time unit generally used elsewhere in ODP. And then the type should be uint64_t to make sure there are enough bits in it.

Reply:
Ack, will fix it in v2.